source/extensions/quic_listeners/quiche/platform/quic_hostname_utils_impl.cc

```diff
@@ -0,0 +1,48 @@
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "extensions/quic_listeners/quiche/platform/quic_hostname_utils_impl.h"
+
+#include <string>
+
+#include "common/http/utility.h"
+
+#include "absl/strings/ascii.h"
+#include "absl/strings/str_cat.h"
+
+// TODO(wub): Implement both functions on top of GoogleUrl, then enable
+// quiche/quic/platform/api/quic_hostname_utils_test.cc.
+
+namespace quic {
+
+// static
+bool QuicHostnameUtilsImpl::IsValidSNI(QuicStringPiece sni) {
+  // TODO(wub): Implement it on top of GoogleUrl, once it is available.
```

Comment:
Fine for now, but note for the SNI check if GoogleUrl takes a while we might be able to that string conversoin to IP fails, and strcat http://hostname using the existing Envoy Utl Utility passes and get full coverage without GoogleUrl

Reply:
Ack. Will definitely do a pass of TODO fixes after all platform impls are populated.